app/permissions/decidim/action_delegator/permissions.rb

```diff
@@ -3,20 +3,30 @@
 module Decidim
   module ActionDelegator
     class Permissions < Decidim::DefaultPermissions
+      SUBJECTS_WHITELIST = [:delegation, :setting, :consultation].freeze
+
       def permissions
         return permission_action unless user.admin?
         return permission_action unless permission_action.scope == :admin
-        return permission_action unless [:delegation, :setting].include?(permission_action.subject)
+        return permission_action unless action_delegator_subject?
 
-        allow! if can_perform_action?(permission_action.action, resource)
+        allow! if consultation_results_exports_action? || can_perform_action?
 
         permission_action
       end
 
       private
 
-      def can_perform_action?(action, resource)
-        if action == :destroy
+      def consultation_results_exports_action?
+        permission_action.subject == :consultation && permission_action.action == :export_results
+      end
+
+      def action_delegator_subject?
+        SUBJECTS_WHITELIST.include?(permission_action.subject)
```

Comment:
well, it's not a big deal in terms of performance to move that array allocation to boot-time rather than request-time but I actually find it more readable that way. You're still reading the `action_delegator_subject?` but the `_WHITELIST` gives you more details about how the method checks that. What's more, the indirection the constant brings makes slightly more changeable IMO.

In any case, this is a micro-optimization not worth spending too much time on.